### PR TITLE
Parallel solution of several LPs in MCLC

### LINKED ISSUE #254: epsilon parameter in MCLC unnecessary
Adding a slack $\epsilon > 0$ is equivalent to requiring $e \ge \epsilon$. But this forces a worse solution by reducing the set of $e$'s over which one minimises. If $e \ge \epsilon$ is a solution it will be found without adding that constraint. If $e \lt \epsilon$ is a solution, then adding the slack variable precludes finding it

### PR DESCRIPTION
**Pull request by mdbenito**
### Description

This PR splits preparation and solution of the linear and quadratic problems in exact and MC least core. This allows batching of the solution of linear problems when running multiple experiments. It also closes #254

### Changes

- Adds `lc_solve_problem`, `lc_solve_problems`, `mclc_prepare_problem`
- Closes #254 

### Checklist

- [x] ~Wrote Unit tests (if necessary)~
- [x] Updated Documentation (if necessary)
- [x] Updated Changelog
- [x] ~If notebooks were added/changed, added boilerplate cells are tagged with `"nbsphinx":"hidden"~

### CODE REVIEW

File: src/pydvl/value/least_core/__init__.py [hunk 1]
```diff
@@ -4,6 +4,70 @@
...
+:func:`~pydvl.value.least_core.montecarlo.mclc_prepare_problem` to prepare a
+list of problems to solve, then solve them in parallel with
+:func:`~pydvl.value.least_core.common.lc_solve_problems`.
+
 """
+from enum import Enum
+from typing import Optional
+
+from ...utils import Utility
+from .. import ValuationResult
```
Comment (AnesBenmerzoug) [inline src/pydvl/value/least_core/__init__.py:22]: I find this one much more readable:

```suggestion
from pydvl.value.results import ValuationResult
```

File: src/pydvl/value/least_core/common.py [hunk 2]
```diff
@@ -9,18 +9,127 @@
 __all__ = [
     "_solve_least_core_linear_program",
     "_solve_egalitarian_least_core_quadratic_program",
+    "lc_solve_problem",
+    "lc_solve_problems",
+    "LeastCoreProblem",
 ]
 
+from pydvl.utils import MapReduceJob, Status, Utility
```
Comment (AnesBenmerzoug) [inline src/pydvl/value/least_core/common.py:17]: These imports should be put before the all dunder variable.

**Comment by AnesBenmerzoug** [comment]
I think it would be good to add tests for the additional functions you defined.

**Review by AnesBenmerzoug — APPROVED** [review]

**Comment by mdbenito** [comment]
@AnesBenmerzoug I think this is good to go now 🤞🏽 after adding some tests as suggested (there were lots of bugs 😅) . We should merge once they run and release 0.4.1 while we wait for all the other stuff to be reviewed

**Comment by AnesBenmerzoug** [comment]
@mdbenito looking at the amount of changes in this and other PRs, I think it would be better to directly release 0.5.0

**Comment by mdbenito** [comment]
> @mdbenito looking at the amount of changes in this and other PRs, I think it would be better to directly release 0.5.0

I suggested it because I don't know how long the rest is going to take, given that everybody is busy. Also, we still have the problem with mismatching docs between dev and what's published...

**Pull request merged**